client/src/main/java/org/eclipse/hono/client/HonoConnection.java

```diff
@@ -240,9 +240,25 @@ static HonoConnection newConnection(final Vertx vertx, final ClientConfigPropert
      *         thus indicates the outcome of executing the code. The future will
      *         be failed with a {@link ServerErrorException} if the <em>context</em>
      *         property is {@code null}.
+     * @deprecated Use {@link #executeOnContext(Handler)} instead.
      */
+    @Deprecated
     <T> Future<T> executeOrRunOnContext(Handler<Future<T>> codeToRun);
 
+    /**
+     * Executes some code on the vert.x Context that has been used to establish the
+     * connection to the peer.
+     * 
+     * @param <T> The type of the result that the code produces.
+     * @param codeToRun The code to execute. The code is required to either complete or
+     *                  fail the promise that is passed into the handler.
+     * @return The future containing the result of the promise passed in to the handler
+     *         for executing the code. The future thus indicates the outcome of executing
+     *         the code. The future will always be failed with a {@link ServerErrorException}
+     *         if this connection's <em>context</em> property is {@code null}.
+     */
+    <T> Future<T> executeOnContext(Handler<Promise<T>> codeToRun);
```

Comment:
Wondering if we can get away without this and just use HonoProtonHelper method directly?

Reply:
This method is mainly here in order to encapsulate the vert.x Context used by the connection. If we remove this method then client code needs to extract the connection's context from the connection before invoking HonoProtonHelper.executeOnContext().